File: src/main/scala/com/abaddon83/burraco/match/games/ports/PlayerPort.scala
package com.abaddon83.burraco.`match`.games.ports

import com.abaddon83.burraco.`match`.games.domainModels.burracoGames.initialised.playerInGames.PlayerInGame
import com.abaddon83.burraco.`match`.games.domainModels.{BurracoPlayer, PlayerNotAssigned}
import com.abaddon83.burraco.shares.players.PlayerIdentity

import scala.concurrent.Future

trait PlayerPort {
  def findPlayerNotAssignedBy(playerIdentity: PlayerIdentity): Future[PlayerNotAssigned]
  def findBurracoPlayerBy(playerIdentity: PlayerIdentity): Future[BurracoPlayer]
  def findBurracoPlayerInGameBy(playerIdentity: PlayerIdentity): Future[PlayerInGame]
}

File: src/main/kotlin/com/abaddon83/burracoGame/commandModel/ports/PlayerPort.kt
package com.abaddon83.burracoGame.commandModel.ports

import com.abaddon83.burracoGame.commandModel.models.BurracoPlayer
import com.abaddon83.burracoGame.commandModel.models.PlayerNotAssigned
import com.abaddon83.burracoGame.commandModel.models.burracoGameExecutions.playerInGames.PlayerInGame
import com.abaddon83.burracoGame.commandModel.models.players.PlayerIdentity

interface PlayerPort {
        suspend fun findPlayerNotAssignedBy(playerIdentity: PlayerIdentity): PlayerNotAssigned
        suspend fun findBurracoPlayerBy(playerIdentity: PlayerIdentity): BurracoPlayer
        suspend fun findBurracoPlayerInGameBy(playerIdentity: PlayerIdentity): PlayerInGame
}